### PR TITLE
Transmutagen

### PR DESCRIPTION
**Pull request by scopatz**
This PR provides a new CRAM-based, code generated generic transmutation solver. This was generated via the [transmutagen](http://www.ergs.sc.edu/transmutagen/) project. 

Since this subsumes decay, the pyne decayer is no longer needed. This simplifies the project significantly, as external downloading and building of pyne's decay functionality has been removed.

The transmutagen solver has a number advantages over what we have in cyclus currently:

* Handles all forms of transmutation, not just decay.
* Runs in ~4 - 8 ms from cold start.
* Faster than ORIGEN by 10 - 1e6 times (due to constant time).
* Faster than other CRAM solvers by 10 - 100 times due to code generation.
* More accurate than other solvers because all computation occur in complex plane.

[Here is a link to some preliminary slides on the method](https://docs.google.com/presentation/d/12hk-Efxe544WJKkdHxnl8dS9qeklRK6CeGyHBkxsTiI/edit?usp=sharing)

We'll be publishing on this shortly, but I wanted to get it out there now that it works. Feel free to ask any questions!

CC @asmeurer @FlanFlanagan

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -159,9 +159,13 @@ FILE(GLOB cc_files "${CMAKE_CURRENT_SOURCE_DIR}/[^_]*.cc")
 SET(CYCLUS_CORE_SRC ${CYCLUS_CORE_SRC}
     ${cc_files}
     "${CMAKE_CURRENT_SOURCE_DIR}/OsiCbcSolverInterface.cpp"
+    "${CMAKE_CURRENT_SOURCE_DIR}/transmute.c"
     )
 
-set_source_files_properties(pyne_decay.cc PROPERTIES COMPILE_FLAGS "-O0")
+add_definitions(-DPYNE_DECAY_IS_DUMMY)
```
Comment (FlanFlanagan) [inline src/CMakeLists.txt]: It looks like you're removing the file that references this, do you still need it?

**Comment by FlanFlanagan** [comment]
Just the one, looks good to me. Might want to update the branch though.

File: src/CMakeLists.txt (hunk 1 above)
Comment (scopatz) [inline src/CMakeLists.txt]: pyne.h and pyne.cc contain the `PYNE_DECAY_IS_DUMMY`, which we did not remove

**Comment by scopatz** [comment]
Updated!

File: src/CMakeLists.txt (hunk 1 above)
Comment (scopatz) [inline src/CMakeLists.txt]: So, yes, this is still needed :)

**Comment by FlanFlanagan** [comment]
Sorry, didn't merge and now it's out of data again.

**Comment by FlanFlanagan** [comment]
I will merge this time! Promise!

**Comment by scopatz** [comment]
Branch updated again.

**Pull request merged**